### PR TITLE
Add managed identity workload identity federation support

### PR DESCRIPTION
**Pull request by dagood**
* Update eng/common to dotnet/docker-tools@947b0c8db5
  * Simple checkout, no need to review.
* Remove unused serviceConnections in eng/common
  * Review this. This means future updates to eng/common need to be careful not to wipe out this change. Included a code comment to make it easier to spot too-early removal in diffs.
  * Unifying our eng/common dirs is tracked by https://github.com/microsoft/go-lab/issues/78.

Test build: https://dev.azure.com/dnceng/internal/_build/results?buildId=2463672&view=results

### CODE REVIEW

File: eng/common/templates/steps/init-docker-linux.yml [hunk 1]
```diff
@@ -34,16 +34,45 @@ steps:
...
-  - script: >
-      echo "##vso[task.setvariable variable=runImageBuilderCmd]
-      docker run --rm
-      -v /var/run/docker.sock:/var/run/docker.sock
-      -v $(Build.ArtifactStagingDirectory):$(artifactsPath)
-      -w /repo
-      $(imageBuilderDockerRunExtraOptions)
-      $(imageNames.imageBuilder.withrepo)"
-    displayName: Define runImageBuilderCmd Variable
+  - task: PowerShell@2
```
Comment (karianna) [inline eng/common/templates/steps/init-docker-linux.yml:37]: Is it better to use a hash commit version than PowerShell@v2 where 2 could represent something different under the hood?  I konwe dependabot supports updating hash commits with a comment e.g. PowerShell@12321qwerrtytughfgf4g4544543 # v2.0.0

File: eng/common/templates/steps/run-imagebuilder.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
+parameters:
```
Comment (karianna) [inline eng/common/templates/steps/run-imagebuilder.yml:1]: General comment that this file could do with some minimal docs

File: eng/common/templates/steps/run-pwsh-with-auth.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
+parameters:
```
Comment (karianna) [inline eng/common/templates/steps/run-pwsh-with-auth.yml:1]: same as above (docs)

File: eng/common/templates/variables/docker-images.yml [hunk 4]
```diff
@@ -1,5 +1,5 @@
 variables:
-  imageNames.imageBuilderName: mcr.microsoft.com/dotnet-buildtools/image-builder:2437925
+  imageNames.imageBuilderName: mcr.microsoft.com/dotnet-buildtools/image-builder:2461919
```
Comment (karianna) [inline eng/common/templates/variables/docker-images.yml:2]: How do we track versions here?

**Review by karianna — APPROVED** [review]

File: eng/common/templates/steps/init-docker-linux.yml (hunk 1 above)
Comment (dagood) [reply eng/common/templates/steps/init-docker-linux.yml:37]: I haven't seen anything like that for AzDO yaml, only GitHub Actions. (https://learn.microsoft.com/en-us/azure/devops/pipelines/process/tasks?view=azure-devops&tabs=yaml#task-versions is all I'm aware of.)

File: eng/common/templates/variables/docker-images.yml (hunk 4 above)
Comment (dagood) [reply eng/common/templates/variables/docker-images.yml:2]: This is updated by a .NET Docker bot upstream: https://github.com/dotnet/docker-tools/commit/947b0c8db5c428f582d9eca7cfefd8abc9c3bd16.

**Review by dagood — COMMENTED** [review]
@karianna it looks like you reviewed the .NET Docker tooling commit. 😅 In general, the code in eng/common is directly from .NET Docker without modification--unless we need to temporarily fix something so it's compatible with our repo, like we have to now. I split it out into its own commit to help differentiate.

**Pull request merged**